### PR TITLE
Udq rst

### PR DESCRIPTION
**Pull request by joakim-hove**
This PR will load UDQ data from the restart file. This is only a first step, and specifically does *not* 

1. Create proper `Schedule::UDQ` datastructures.
2. Consider `UDA` usage.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Comments?

File: opm/io/eclipse/rst/udq.hpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    void add_well_value(const std::string& wname, double value);
+    void add_group_value(const std::string& wname, double value);
+    void add_field_value(double value);
+
+    std::string name;
+    std::string unit;
+    UDQVarType var_type;
+    std::optional<std::string> define;
+    std::vector<std::pair<std::string, double>> well_values;
+    std::vector<std::pair<std::string, double>> group_values;
```
Comment (bska) [inline opm/io/eclipse/rst/udq.hpp:51]: You strictly speaking need `<utility>` in order to bring `std::pair` into scope.

File: src/opm/io/eclipse/rst/state.cpp [hunk 2]
```diff
@@ -174,6 +176,55 @@ void RstState::add_msw(const std::vector<std::string>& zwel,
...
+        const auto& name = zudn[udq_index * UDQDims::entriesPerZUDN()];
+        const auto& unit = zudn[udq_index * UDQDims::entriesPerZUDN() + 1];
+        auto define_int =  iudq[udq_index * UDQDims::entriesPerIUDQ()];
+
+        if (define_int == 2) {
+            auto zudl_begin = zudl.begin();
+            auto zudl_end = zudl.begin();
+            std::advance( zudl_begin, udq_index * UDQDims::entriesPerZUDL() );
+            std::advance( zudl_end, (udq_index + 1) * UDQDims::entriesPerZUDL() );
+            auto udq_define = std::accumulate(zudl_begin, zudl_end, std::string{}, std::plus<std::string>());
```
Comment (bska) [inline src/opm/io/eclipse/rst/state.cpp:196]: You need `<numeric>` to bring a declaration of `std::accumulate()` into scope.

File: src/opm/io/eclipse/rst/state.cpp [hunk 3]
```diff
@@ -174,6 +176,55 @@ void RstState::add_msw(const std::vector<std::string>& zwel,
...
+            if (udq_define[0] == '~')
+                udq_define[0] = '-';
+
+            this->udqs.emplace_back(name, unit, udq_define);
+        } else
+            this->udqs.emplace_back(name, unit);
+
+        auto& udq = this->udqs.back();
+        if (udq.var_type == UDQVarType::WELL_VAR) {
+            for (std::size_t well_index = 0; well_index < this->wells.size(); well_index++) {
```
Comment (bska) [inline src/opm/io/eclipse/rst/state.cpp:206]: You need `<cstddef>` (or one of a few [other headers](https://en.cppreference.com/w/cpp/types/size_t)) to bring a declaration of `std::size_t` into scope.

**Review by bska — COMMENTED** [review]
Looks good except for a few missing headers.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**